tests/integration_tests/tests/agentless_tests/test_snapshot.py

```diff
@@ -306,32 +306,36 @@ def _get_snapshot(self, name):
 
     def _upload_and_restore_snapshot(self,
                                      snapshot_path,
-                                     tenant_name=None,
+                                     tenant_name=DEFAULT_TENANT_NAME,
                                      snapshot_id=None):
         """Upload the snapshot and launch the restore workflow
         """
         snapshot_id = snapshot_id or self.SNAPSHOT_ID
-        self._upload_and_validate_snapshot(snapshot_path, snapshot_id)
+        rest_client = utils.create_rest_client(tenant=tenant_name)
+        self._upload_and_validate_snapshot(snapshot_path,
+                                           snapshot_id,
+                                           rest_client)
         self.logger.debug('Restoring snapshot...')
-        execution = self.client.snapshots.restore(
-            snapshot_id,
-            tenant_name=tenant_name
-        )
-        execution = self._wait_for_execution_to_end(execution)
+        execution = rest_client.snapshots.restore(snapshot_id)
+        execution = self._wait_for_execution_to_end(execution, rest_client)
         if execution.status == Execution.FAILED:
             self.logger.error('Execution error: {0}'.format(execution.error))
         self.assertEqual(Execution.TERMINATED, execution.status)
 
-    def _upload_and_validate_snapshot(self, snapshot_path, snapshot_id):
+    def _upload_and_validate_snapshot(self,
+                                      snapshot_path,
+                                      snapshot_id,
+                                      rest_client=None):
```

Comment:
The default value "None" is not necessary (because the function is only called once, with this parameter supplied) - and if it was necessary the case when it is None should be treated.